### PR TITLE
add log clear command with time offset option

### PR DESCRIPTION
**Pull request by ThomasDeMarez**
This PR adds the `schedules/clear-logs` command, optionally a time offset can be supplied using the `offset` param. This way it is possible to keep recent logs and discart of old ones eg; `schedules/clear-logs --offset=12h`.

This command could be triggered using the scheduler itself in order to keep the logs table from exponentially growing.

### CODE REVIEW

**Comment by panlatent** [comment]
Good job! Thanks!

**Comment by panlatent** [comment]
@ThomasDeMarez  `--offset` is a misleading option name. This option seems to be more suitable for using time-related name.  Maybe, `before` .... i think you'll come up with something better name.

**Comment by ThomasDeMarez** [comment]
Could be `before` or `age` or maybe a hyphened param like `older-than`?
Let me know what you want it to be called.

**Comment by panlatent** [comment]
`schedules/clear-logs` means **DELETE ALL** `schedules/clear-logs --all`
`schedules/clear-logs --offset=12h` means **DELETE BUT exclude last 12 hours**  `delete(init, now - 12h)`

`keep-time` ... `exclude` ....

I gave up the previous name. Rethinking, our purpose is to keep recent effective logs, the command `clear-logs` indicates to clear all, we need an option name that clearly indicates `exclude` and the data format here is time.

 I think the option name `expire` should be better, it satisfies the above two requirements without ambiguity.

```bash
$ craft schedules/clear-logs --expire=12h
$ craft schedules/clear-logs --expire=30d
```

This is by far the name I'm more than happy with. I think it is possible to support this parameter in the plugin settings, or even call it automatically. If so, then `clear-logs` meas `clear-logs --expire=<setting-value>` , add the `--all` option.

**Comment by ThomasDeMarez** [comment]
@panlatent I've updated the commit as requested.

- There now is a Log setting where the user can provide a default expiry. This is also tied to a CarbonInterval validator
- The clear-logs command now checks for a default expire in the settings, but does allow the expire to be overwritten when te param is added to the command
- The clear-logs command has a `--all` option to clear all logs

File: src/translations/zh/schedule.php [hunk 1]
```diff
@@ -111,4 +111,6 @@
     'Add Header Value' => '添加首部值',
     'Add URL Parameter' => '添加 URL 参数',
     'Add Value' => '添加值',
-];
\ No newline at end of file
+    'Expiry threshold' => '????',
+    'Default expiry time used when calling the `schedules/clear-logs` command.' => '???? `schedules/clear-logs`',
```
Comment (ThomasDeMarez) [inline src/translations/zh/schedule.php:115]: Don't know the translations for these strings

**Comment by ThomasDeMarez** [comment]
Thanks for the merge! When can we expect a new release including these changes?
And is there an option for a Craft 3 release? We've got Craft 3 sites that could use this as wel.

**Pull request merged**